weave/flow/eval.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+import time
+import inspect
+import traceback
+import typing
+from typing import Any, Callable, Optional, Union
+import numpy as np
+
+import weave
+from weave import op_def
+from weave.flow import Object, Dataset, Model
+from weave.flow.scorer import Scorer, get_scorer_attributes, auto_summarize
+from weave.weaveflow import util
+
+from rich.console import Console
+from rich import print
+
+console = Console()
+
+
+def async_call(
+    func: typing.Union[Callable, op_def.OpDef], *args: Any, **kwargs: Any
+) -> typing.Coroutine:
+    is_async = False
+    if isinstance(func, op_def.OpDef):
+        is_async = inspect.iscoroutinefunction(func.raw_resolve_fn)
+    else:
+        is_async = inspect.iscoroutinefunction(func)
+    if is_async:
+        return func(*args, **kwargs)  # type: ignore
+    else:
+        return asyncio.to_thread(func, *args, **kwargs)
+
+
+class Evaluation(Object):
+    dataset: Union[Dataset, list]
+    scorers: Optional[list[Union[Callable, op_def.OpDef, Scorer]]] = None
+    preprocess_model_input: Optional[Callable] = None
+
+    def model_post_init(self, __context: Any) -> None:
+        scorers = []
+        for scorer in self.scorers or []:
+            if isinstance(scorer, Scorer):
+                pass
+            elif callable(scorer) and not isinstance(scorer, op_def.OpDef):
+                scorer = weave.op()(scorer)
+            elif isinstance(scorer, op_def.OpDef):
+                pass
+            else:
+                raise ValueError(f"Invalid scorer: {scorer}")
+            scorers.append(scorer)
+        self.scorers = scorers
+
+        if isinstance(self.dataset, list):
+            self.dataset = Dataset(rows=self.dataset)
+
+        if self.name == None and self.dataset.name != None:
```

Comment:
Lots of comparisons with None that I think should be `is None`/`is not None`

Reply:
Sadly, we have to do it this way in Weave. See my other response.